xAPISGDataGenerator.ipynb

```diff
@@ -0,0 +1,705 @@
+    "\n",
+    "template_result_action_type_interact={\n",
+    "    \"extensions\": {\n",
+    "        \"action_type\": None,\n",
+    "    }\n",
+    "}\n",
+    "\n",
+    "def generate_GameObjects(change):\n",
+    "    start_time=datetime.now()\n",
+    "    with output:\n",
+    "        print(\"Starting the data generation\")\n",
+    "    verbs=checkboxes_verbs_gameObjects.get_selected_options()\n",
+    "    types=checkboxes_types_gameObjects.get_selected_options()\n",
+    "    actionTypes=checkboxes_action_types.get_selected_options()\n",
+    "    ids=generate_IDs()\n",
+    "    times=generate_times_datas()\n",
+    "    player=generate_players()\n",
+    "    statements=[]\n",
+    "    checkTypes={}\n",
+    "    checkActionType={}\n",
+    "    for i in range(nbtraces.value):\n",
+    "        randomID=random.choice(ids)\n",
+    "        randomPlayer=random.choice(player)\n",
+    "        randomVerb=random.choice(verbs)\n",
+    "        randomType= checkType(checkTypes,randomID,randomPlayer,types)\n",
+    "        time=times[i]\n",
+    "        statement=generator(randomPlayer, randomVerb, randomType,randomID, time)\n",
+    "        if randomVerb==\"interacted\":\n",
+    "            if randomType not in checkActionType.keys():\n",
+    "                checkActionType[randomID]=random.choice([True, False])\n",
+    "            generateActionType=checkActionType[randomID]\n",
+    "            if generateActionType and not len(actionTypes)==0:\n",
+    "                actiontype=random.choice(actionTypes)\n",
+    "                if actiontype==\"custom\":\n",
+    "                    actiontype=str(uuid.uuid4())\n",
+    "                    actionTypes.append(actiontype)\n",
+    "                result_action_type=copy.deepcopy(template_result_action_type_interact)\n",
+    "                result_action_type[\"extensions\"][\"action_type\"]=actiontype\n",
+    "                statement[\"result\"]=result_action_type\n",
+    "        statements.append(statement)\n",
+    "    save_generated_data(statements, verbs, types)\n",
+    "    end_time=datetime.now()\n",
+    "    delta=end_time-start_time\n",
+    "    with output:\n",
+    "        print(\"End the data generation\")\n",
+    "        print(\"Generated in \"+str(delta))\n",
+    "        clear_output(wait=True)\n",
+    "\n",
+    "buttonGameObjects=widgets.Button(description='Generate Data')\n",
+    "buttonGameObjects.on_click(generate_GameObjects)\n",
+    "\n",
+    "tabgameObjects=VBox([\n",
+    "    HBox([\n",
+    "        checkboxes_verbs_gameObjects.multi_select,\n",
+    "        checkboxes_types_gameObjects.multi_select,\n",
+    "        checkboxes_action_types.multi_select\n",
+    "    ]),\n",
+    "    HBox([\n",
+    "        vbox_nb_traces_ID_player,\n",
+    "        buttonGameObjects\n",
+    "    ])\n",
+    "])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**User interface**"
```

Comment:
Possible improvement: Could some of the previous code be hidden in another notebook, so this main one is left with the user interface (as simple as possible)?